bazel/execution/service.go

```diff
@@ -98,141 +98,120 @@ func (s *executionServer) Execute(
+// google LongRunning Operation message
 func (s *executionServer) GetOperation(
-	ctx context.Context,
+	_ context.Context,
 	req *longrunning.GetOperationRequest) (*longrunning.Operation, error) {
-	log.Infof("Received GetOperation request: %s", req)
+	log.Infof("Received GetOperation request: %v", req)
 
 	if !s.IsInitialized() {
 		return nil, status.Error(codes.Internal, "Server not initialized")
 	}
 
-	// TODO get status of job identified by req.Name
+	rs, err := s.getRunStatusAndValidate(req.Name)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
 
-	op := longrunning.Operation{}
-	op.Name = req.Name
-	op.Done = true
+	actionResult := bazelapi.MakeActionResultDomainFromThrift(rs.GetBazelResult())
 
 	eom := &remoteexecution.ExecuteOperationMetadata{
-		Stage: remoteexecution.ExecuteOperationMetadata_COMPLETED,
+		Stage:        runStatusToExecuteOperationMetadata_Stage(rs),
+		ActionDigest: actionResult.GetActionDigest(),
 	}
 
 	// Marshal ExecuteActionMetadata to protobuf.Any format
 	eomAsPBAny, err := marshalAny(eom)
 	if err != nil {
-		return nil, err
+		return nil, status.Error(codes.Internal, err.Error())
 	}
-	op.Metadata = eomAsPBAny
 
 	// Marshal ExecuteResponse to protobuf.Any format
 	res := &remoteexecution.ExecuteResponse{
-		Result: &remoteexecution.ActionResult{
-			ExitCode: 0,
-		},
+		Result:       actionResult.GetResult(),
 		CachedResult: false,
+		Status:       runStatusToGoogleRpcStatus(rs),
 	}
-
 	resAsPBAny, err := marshalAny(res)
 	if err != nil {
-		return nil, err
+		return nil, status.Error(codes.Internal, err.Error())
 	}
 
 	log.Info("GetOperationRequest completed successfully")
 	// Include the response message in the longrunning operation message
-	op.Result = &longrunning.Operation_Response{Response: resAsPBAny}
+	op := longrunning.Operation{
+		Name:     req.Name,
+		Metadata: eomAsPBAny,
+		Done:     runStatusToDoneBool(rs),
+		Result: &longrunning.Operation_Response{
+			Response: resAsPBAny,
+		},
+	}
 	return &op, nil
 }
 
-func marshalAny(pb proto.Message) (*any.Any, error) {
-	pbAny, err := ptypes.MarshalAny(pb)
+func (s *executionServer) ListOperations(context.Context, *longrunning.ListOperationsRequest) (*longrunning.ListOperationsResponse, error) {
+	return nil, status.Error(codes.Internal, fmt.Sprint("Not implemented"))
+}
+
+func (s *executionServer) DeleteOperation(context.Context, *longrunning.DeleteOperationRequest) (*empty.Empty, error) {
+	return nil, status.Error(codes.Internal, fmt.Sprint("Not implemented"))
+}
+
+func (s *executionServer) CancelOperation(context.Context, *longrunning.CancelOperationRequest) (*empty.Empty, error) {
+	return nil, status.Error(codes.Internal, fmt.Sprint("Not implemented"))
+}
+
+func (s *executionServer) getRunStatusAndValidate(jobID string) (*runStatus, error) {
+	js, err := api.GetJobStatus(jobID, s.sagaCoord)
+	if err != nil {
+		return nil, status.Error(codes.Internal, fmt.Sprintf("Error getting job status: %s", err))
```

Comment:
should be fine with 'normal' errors here since top-level is wrapping into a status.Error